.github/workflows/main.yml

```diff
@@ -1,15 +1,59 @@
-name: Ubuntu
+name: CI
 
-on: [push]
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
 
 jobs:
-  build:
-
-    runs-on: ubuntu-18.04
+  test:
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        python-version: [3.6]
+        python-version: [pypy3, 2.7, 3.5, 3.6, 3.7, 3.8]
+        os: [macos-10.15, ubuntu-18.04, ubuntu-20.04, windows-2019]
+        include:
+          - os: ubuntu-18.04
+            python-version: pypy2
+          - os: ubuntu-20.04
+            python-version: pypy2
+            experimental: true
```

Comment:
Why is this experimental?

Reply:
Ah, this was before I added `fail-fast: false` as I didn't want this failure to stop the entire build. I'll remove it.